getnacelle/nuxt-reference-store
# feat: enable Shopify checkout from cart

### WHY are these changes introduced?

Addresses [LD-1236](https://nacelle.atlassian.net/browse/LD-1236)

> Add @nacelle/shopify-checkout to reference store

Before this PR, clicking the cart's **Checkout** button didn't create a checkout.

### WHAT is this pull request doing?

Adds Shopify checkout functionality from [`@nacelle/shopify-checkout`](https://www.npmjs.com/package/@nacelle/shopify-checkout).

## 

### How to Test

- Go to https://nuxt-reference-store-git-ld-1236-checkout-from-cart-nacelle.vercel.app
- Add some items to the cart and perform a checkout
  - the `checkoutId` should be saved to `localStorage`
  - if the checkout isn't completed, the same `checkoutId` should be reused, even if new products are added to the cart in between checkouts
  - if the checkout is completed, the cart should reset

### Demonstration

#### Creating a checkout

https://user-images.githubusercontent.com/5732000/137990081-a8f1ab1f-e2aa-4d66-9a2d-27bf162507ba.mov

#### A completed checkout clears the cart

https://user-images.githubusercontent.com/5732000/137990157-90755b3a-ef24-40ee-ac2b-2a0f720f74c0.mov
